### ISSUE TITLE
Feat: show onboarding button on small screens 

### ISSUE THREAD

**Issue by a0m0rajab** [open-sauced/app#1462]
# Current behavior: 

The small screens does not show the onboarding button 

# Suggestions: 

              > I think such a design would be better than just hiding the element: 
> 
> ![image](https://github.com/open-sauced/insights/assets/18273833/89e57c15-2784-4f16-903b-b3014ea9dc8b)
> 
> 

I do like this design. Feel free to open a new issue removing the hidden flag on mobile.

_Originally posted by @bdougie in https://github.com/open-sauced/insights/issues/1460#issuecomment-1660334242_

**Comment by dev-phantom** [comment]
.take

**Comment by dev-phantom** [comment]
I appreciate the opportunity to contribute to this project. However, I regret to inform you that I won't be able to fix this issue at the moment. The project relies on Next.js, and my current processor is a dual-core, which limits my ability to effectively work with Next.js projects, especially when dealing with a large codebase.

I apologize for any inconvenience this may cause and hope that someone else with better hardware can address this issue.

**Comment by takanome-dev** [comment]
Tried doing this but in some pages it doesn't look good. Maybe it should stay hidden for mobile for now

![2023-09-19_10-00](https://github.com/open-sauced/app/assets/79809121/3583f7dc-9c28-4c64-a251-b02d66d70949)

**Comment by dev-phantom** [comment]
> Tried doing this but in some pages it doesn't look good. Maybe it should stay hidden for mobile for now
> 
> ![2023-09-19_10-00](https://user-images.githubusercontent.com/79809121/268918500-3583f7dc-9c28-4c64-a251-b02d66d70949.png)

I feel on this page you should fill the button background with orange and it should be before the cover image 

Can you create a draft PR on this?

**Comment by takanome-dev** [comment]
> I feel on this page you should fill the button background with orange and it should be before the cover image

I doubt changing the design of the pages to make this work is the right way to do it. Would love to hear from @open-sauced/engineering on this before proceeding

**Comment by nickytonline** [comment]
We need to take a look at mobile as a whole as opposed to one screen at a time. I'm going to sync up with @isabensusan.

**Comment by babblebey** [comment]
> > Tried doing this but in some pages it doesn't look good. Maybe it should stay hidden for mobile for now
> > 
> > ![2023-09-19_10-00](https://user-images.githubusercontent.com/79809121/268918500-3583f7dc-9c28-4c64-a251-b02d66d70949.png)
> 
> I feel on this page you should fill the button background with orange and it should be before the cover image 
> 
> Can you create a draft PR on this?

This will no longer be the case, because I did a quick refactor on the `TopNav` component in #1678, it now separates the Nav displayed on Mobile/SM Devices from what is displayed on Desktops/MD devices.

Appending this button now will basically be adding the button to the mobile part of it.

**Comment by 289Volts** [comment]
.take

**Comment by takanome-dev** [comment]
Hey @289Volts, thanks for your interest in working on this.

Let's wait and hear first from @nickytonline before you start working on this.

**Comment by 289Volts** [comment]
> Hey @289Volts, thanks for your interest in working on this.
> 
> 
> 
> Let's wait and hear first from @nickytonline before you start working on this.

Okay no problem

**Comment by babblebey** [comment]
> We need to take a look at mobile as a whole as opposed to one screen at a time. I'm going to sync up with isabensusan.

Hey @nickytonline,

I wanted to point that right this moment and since #1460, the Onboarding button has been absent on mobile/small screen devices.

Considering that `Onboarding` is an imperative step to unlocking the `Insight` page for a new user, and noting the fact that traffic to websites are mostly generated from mobile devices, I think we've/we'll been doing the project a dis-service by not addressing this screen asap now for the multi-tude of mobile users not getting the full experience now.

We can wait on @isabensusan input, but it might take a while to conclude for the mobile experience in whole now.

I have a design idea, if @289Volts can implement this, I think we will be prime to address this almost permanently.

See Screenshot below...

![IMG_20231002_000447.jpg](https://github.com/open-sauced/app/assets/25631971/d7bb2091-1903-47bf-8872-a59102445e43)

### Explainer

- Display the onboarding button with only the icon on Mobile.
- Transfer the text to a tooltip that renders the exact text that was removed for mobile i.e. "Complete the onboarding"
- This button should be positioned in the mobile Nav in a flex box with `space-between` justification to let it stick opposite the Nav items.

I honestly don't enjoy seeing mobile experiences lacking for a single second at all 👀

🍕

**Comment by 289Volts** [comment]
> > We need to take a look at mobile as a whole as opposed to one screen at a time. I'm going to sync up with isabensusan.
> 
> Hey @nickytonline,
> 
> I wanted to point that right this moment and since #1460, the Onboarding button has been absent on mobile/small screen devices.
> 
> Considering that `Onboarding` is an imperative step to unlocking the `Insight` page for a new user, and noting the fact that traffic to websites are mostly generated from mobile devices, I think we've/we'll been doing the project a dis-service by not addressing this screen asap now for the multi-tude of mobile users not getting the full experience now.
> 
> We can wait on @isabensusan input, but it might take a while to conclude for the mobile experience in whole now.
> 
> I have a design idea, if @289Volts can implement this, I think we will be prime to address this almost permanently.
> 
> See Screenshot below...
> 
> ![IMG_20231002_000447.jpg](https://user-images.githubusercontent.com/25631971/271858205-d7bb2091-1903-47bf-8872-a59102445e43.jpg)
> 
> ### Explainer
> * Display the onboarding button with only the icon on Mobile.
> * Transfer the text to a tooltip that renders the exact text that was removed for mobile i.e. "Complete the onboarding"
> * This button should be positioned in the mobile Nav in a flex box with `space-between` justification to let it stick opposite the Nav items.

https://github.com/open-sauced/app/assets/66239216/79fee6db-5900-4bea-8a91-ae209878b207


> 
> I honestly don't enjoy seeing mobile experiences lacking for a single second at all 👀
> 
> 🍕

do you mean something like this? 
https://github.com/open-sauced/app/assets/66239216/79fee6db-5900-4bea-8a91-ae209878b207

Took the initiative to spin something up. The button is hidden after it crosses 640px which is the 'sm' media query in TailwindCSS. I also made it accessible to the button by adding an aria-labelledby attribute to the button so that screen readers will know that the pie is a prompt to complete the onboarding process. 

Do let me know your thoughts on this

**Comment by babblebey** [comment]
> do you mean something like this?

That's it! 👍🏾

You already implemented that!? 👀

Well, now you wait for @nickytonline's input 🤞🏾

If you will be submitting a PR for this, you might need a demo that shows your decreasing/increasing your screen-size so to show how the button behaves at all breakpoints.

**Comment by 289Volts** [comment]
> > do you mean something like this?
> 
> That's it! 👍🏾
> 
> You already implemented that!? 👀
> 
> Well, now you wait for @nickytonline's input 🤞🏾
> 
> If you will be submitting a PR for this, you might need a demo that shows your decreasing/increasing your screen-size so to show how the button behaves at all breakpoints.

After implementation, this is what it looks like on different breakpoints

https://github.com/open-sauced/app/assets/66239216/23021d5d-3a1f-4b9b-9b62-273c5ef815d0
